docbrown/src/core/tgraph.rs

```diff
@@ -883,7 +884,140 @@ impl TemporalGraph {
             })
             .collect()
     }
-}
+
+    pub(crate) fn temporal_vertex_timestamps_vec(&self, src: u64) -> Vec<i64> {
+        let src_pid =  self.logical_to_physical[&src];
+        self.timestamps[src_pid].iter().map(|t| *t).collect()
+    }   
+
+    pub(crate) fn temporal_vertex_timestamps_vec_window(&self, src: u64, w: Range<i64>) -> Vec<i64> {
+        let src_pid = self.logical_to_physical[&src];
+        self.timestamps[src_pid].range(w).map(|t| *t).collect()
+    }
+
+    pub(crate) fn temporal_edge_timestamps_vec(
+        &self,
+        src: u64,
+        layer: usize,
+        d: Direction
+    ) -> Vec<i64> {
+        let mut list: Vec<i64> = Vec::new();
+        let src_pid = self.logical_to_physical[&src];
+        if d == Direction::IN {
+            match self.layers[layer].adj_lists.get(src_pid).unwrap_or(&Adj::Solo) {
+                Adj::Solo => list,
+                Adj::List {into, ..} => {
+                    list.push(into.find(src_pid).map(|e| e.0).unwrap_or(0));
```

Comment:
I think we probably want to filter this down or clone it into a vet instead of pushing individual values into a new vector, will be faster.